### ISSUE TITLE
Add a `gm reboot` command

### ISSUE THREAD

**Issue by seanchen1991** [informalsystems/gm#3]
## Summary

Add a `gm reboot` command that resets the state of `gm`.

## Problem Definition

Sometimes the state of the local networks and full nodes on my machine get pretty messed up. In that scenario, the easiest fix is to simply reboot everything to a blank slate and start over. 

Currently, this requires manually removing the appropriate directories. It would be handy to add a `gm` command to do this more quickly and easily.

## Proposal

Add a `gm reboot` command that resets the state of `gm`. This command would:
1. Remove all full nodes and networks from `/.gm`
2. Run `killall gaiad`

## Acceptance Criteria

- [ ] `gm reboot` command added that works as expected

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by greg-szabo** [comment]
There's a `rm` command that removes the folders. So, what you're looking for is `gm stop && gm rm && gm start`?

The "problem" with the `rm` command is that it's very cautious: the command-line requires input names so you don't accidentally delete all your config with `gm rm`.

But we could have a modified version of it calling `stop/rm/start`.

**Comment by seanchen1991** [comment]
> `gm stop && gm rm && gm start`
Yes, I think this is essentially what I had in mind when 🙂

**Comment by greg-szabo** [comment]
My concern is the naming convention. `restart`, `reboot` and `reset` are overloaded words and they are used differently in the Cosmos SDK than you'd normally expect. Reset in usual computer terminology should mean something that does a non-destructive cold start of the process, something like stop any processes, clear caches (possibly by unpowering them for a moment) and start processes again. However, in Cosmos SDK terms reset means a destructive event: stop all processes, remove any database (work) that has been done (while keeping the configuration around) and start it again.

So the current process will stop any process, remove all work (database) that has been done and then scrub the configuration out of existence. Then it will recreate the configuration from scratch based on the config file and start the (possibly very different) processes again. The end state can be fundamentally different from the start state.

Alternative naming idea for the command:
`gm scrub`.

**Comment by seanchen1991** [comment]
I defer to whatever name you think is best 😄

**Comment by greg-szabo** [comment]
Also, `gm` is only aware of processes running that are defined in the config file. A complete "reboot" of what is on the host machine is something else. So I need to understand the motivation behind the command. Do you need this command so you can clear out what's on the host machine or do you need this because you want to restart your networks from scratch (eg: because they are above a height, or the genesis changed)?

Maybe `gm scrub` should be a host command that goes through ports 27000-28000 (or whatever is defined) and kills all processes there.
Then `gm reconfig` could be a command that stops all processes and deletes config, then recreates config.

**Comment by seanchen1991** [comment]
If I recall correctly, the main motivation for opening this issue was one of convenience, i.e. I found myself needing to reset the state of my local network back to the configured state enough times that I figured a single command that alias'd `gm stop && gm rm && gm start` would be useful; that's about it. 

It sounds like a hypothetical `gm scrub` command would have implications on use-cases that I am not aware of that need to be considered. In light of that, I'm not sure if this quick-fix idea is worthwhile.
